puddlestuff/puddleobjects.py

```diff
@@ -654,20 +654,26 @@ def getfiles(files, subfolders=False):
             if not isdir(f):
                 yield f
             else:
-                dirname, subs, fnames = next(os.walk(f))
-                for fname in fnames:
-                    yield join(dirname, fname)
+                try:
+                    dirname, subs, fnames = next(os.walk(f))
```

Comment:
im sorry, i dont think this is the right approach. `for` loops already exit upon a StopIteration exception being raised. so i think what this PR needs is to mimik what L665-670 were before:

```
for dirname, subs, fnames in os.walk(f):
    do work
```

Reply:
It works and is much better, thanks.